Making Sense Project API/Controllers/CarController.cs

```diff
@@ -16,67 +16,55 @@ public CarController(ICarCRUD<Car> carCRUD)
             _carCRUD = carCRUD;
         }
 
-        [HttpGet("byId")]
-        public IActionResult Get(int idCar)
+        [HttpPost("Create")]
+        public IActionResult Create(Car car)
         {
-            Car car;
-            try
-            {
-                car = _carCRUD.GetById(idCar);
-            }
-            catch (System.Exception ex)
+            if (_carCRUD.GetById(car.IdCar) != null)
             {
-                return BadRequest(ex.Message);
+                return BadRequest($"Ya existe un auto con id {car.IdCar} creado");
             }
-            return Ok(car);
+            _carCRUD.Create(car);
+            return Ok();
         }
 
-        [HttpGet("All")]
-        public IActionResult GetAll()
+        [HttpPut("Update")]
+        public IActionResult Update(Car car)
         {
-            IList<Car> list = _carCRUD.GetAll();
-            return Ok(list.OrderBy(x => x.IdCar));
+            if (_carCRUD.GetById(car.IdCar) == null)
+            {
+                return NotFound($"No se encontro auto con id {car.IdCar} para actualizar");
+            }
+            _carCRUD.Update(car);
+            return Ok();
         }
 
         [HttpDelete]
         public IActionResult Delete(int idCar)
         {
-            try
-            {
-                _carCRUD.DeleteById(idCar);
-            }
-            catch (System.Exception ex)
+            if (_carCRUD.GetById(idCar) == null)
             {
-                return BadRequest(ex.Message);
+                return NotFound($"No se encontro auto con id {idCar} para eliminar");
             }
+            _carCRUD.DeleteById(idCar);
             return Ok();
         }
-        [HttpPost("Create")]
-        public IActionResult Create(Car car)
+
+        [HttpGet("byId")]
+        public IActionResult Get(int idCar)
         {
-            try
-            {
-                _carCRUD.Create(car);
-            }
-            catch (System.Exception ex)
+
+            if (_carCRUD.GetById(idCar) == null)
             {
-                return BadRequest(ex.Message);
+                return NotFound($"No se encontro auto con id {idCar} registrado");
             }
-            return Ok();
+            return Ok(_carCRUD.GetById(idCar));
         }
-        [HttpPut("Update")]
-        public IActionResult Update(Car car)
+
+        [HttpGet("All")]
+        public IActionResult GetAll()
         {
-            try
-            {
-                _carCRUD.Update(car);
-            }
-            catch (System.Exception ex)
-            {
-                return BadRequest(ex.Message);
-            }
-            return Ok();
+            IList<Car> list = _carCRUD.GetAll();
```

Comment:
Qué te parece un nombre mas representativo? 
```suggestion
            IList<Car> cars = _carCRUD.GetAll();
```
Además, para seguir con convenciones:
```suggestion
            var cars = _carCRUD.GetAll();
```